### PR TITLE
Payment Process Modal 

### PR DESCRIPTION
**Pull request by jarodpeachey**
### What github.com/wevote/WebApp/issues does this fix?
#2451 
### Changes included this pull request?

Sorry this took so long!

This covers the design of the first 3 stages of the payment process, excluding the final one (the payment summary stage). 

I added test coupon codes to the component state in an array to simulate having multiple promos running at once. I also added the validForProfessionalPlan and validForEnterprisePlan booleans to these coupon objects to simulate some coupons only being valid for one plan, and not the other.

I also made sure that the monthly and annual plan prices will update based on if the user goes back and changes their plan on the first screen. I made sure that the data stays on screen resize as well, and switches between desktop and mobile layout without losing the state.

Let me know if this is good to go, or if I should design the last page before you approve this. I'll demo it during the standup tomorrow so everyone can see it, and we can decide then.

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -20,14 +20,15 @@
     <meta property="og:title"         content="We Vote" />
     <meta property="og:description"   content="View your ballot. Learn from friends. Share your vision." />
     <meta property="og:image"         content="https://WeVote.US/img/global/logos/we-vote-logo-wordmark-vertical-inverted-color-256x256.png" />
-    <link href="/css/loading-screen.css" rel="stylesheet" type="text/css" />
+    <base href="/">
+    <link href="/css/loading-screen.css" type="text/css" />
 
-    <link rel="apple-touch-icon" sizes="180x180" href="/img/global/icons/apple-touch-icon.png">
```
Comment (DaleMcGrew) [inline src/index.html]: Hi @jarodpeachey, thank you for this pull request! I wanted to check in about the removal of the "link rel=" attributes. I believe we still want these "rel" tags. What is the benefit of removing them?

Comment (jarodpeachey) [reply src/index.html]: I can double-check, but I believe these helped with making sure that Webpack loaded the styles properly. Once I removed these, the MIME type error no longer showed in the console.

Like I said, I'll double-check and get back to you before or during the standup. 👍

**Comment by jarodpeachey** [comment]
@DaleMcGrew Here's the comparison of the console before and after I remove the rel attributes:

### **With rel attributes**
with_rel


### **Without rel attributes**
without_rel

**Comment by DaleMcGrew** [comment]
Thank you @jarodpeachey this analysis is very helpful. I will roll back the index.html changes, but create a new issue for the team working on webpack, since this is an interesting clue. 👍

**Pull request merged**